File: system-tests-fixtures/api/ComplexApi/android/lib/src/main/java/com/ComplexApi/ern/api/SysteTestEventApi.java
/*
* Copyright 2017 WalmartLabs
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* http://www.apache.org/licenses/LICENSE-2.0
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

package com.ComplexApi.ern.api;

import android.support.annotation.NonNull;

import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEventListener;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEvent;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeRequestHandler;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeResponseListener;
import com.walmartlabs.electrode.reactnative.bridge.None;
import java.util.*;
import java.util.UUID;


public final class SysteTestEventApi {
    private static final Requests REQUESTS;
    private static final Events EVENTS;

    static {
        REQUESTS = new SysteTestEventRequests();
        EVENTS = new SysteTestEventEvents();
    }

    private SysteTestEventApi() {
    }

    @NonNull
    public static Requests requests() {
        return REQUESTS;
    }

    @NonNull
    public static Events events() {
        return EVENTS;
    }

    public interface Events {
        String EVENT_TEST_EVENT = "com.ComplexApi.ern.api.event.testEvent";

        UUID addTestEventEventListener(@NonNull final ElectrodeBridgeEventListener<String> eventListener);

            ElectrodeBridgeEventListener<ElectrodeBridgeEvent> removeTestEventEventListener(@NonNull final UUID uuid);

        void emitTestEvent(@NonNull String buttonId);

    }

    public interface Requests {


    }
}

File: system-tests-fixtures/api/ComplexApi/IOS/SysteTestEventAPI.swift
public class SysteTestEventAPI: NSObject  {
    static let kEventTestEvent = "com.ComplexApi.ern.api.event.testEvent";


    public lazy var events: SysteTestEventAPIEvents = {
        return SysteTestEventEvents()
    }()


    public lazy var requests: SysteTestEventAPIRequests = {
        return SysteTestEventRequests()
    }()
}

public class SysteTestEventAPIEvents: NSObject {
    public func addTestEventEventListener(eventListener: @escaping ElectrodeBridgeEventListener) {
        assertionFailure("should override")
    }

    public func emitEventTestEvent(buttonId: String) {
        assertionFailure("should override")

    }
}

public class SysteTestEventAPIRequests: NSObject {
}

